### ISSUE TITLE
[4.0] Cannot disable the X-Content-Type-Options in the System - HTTP Headers plugin

### ISSUE THREAD

**Issue by sandewt** [joomla/joomla-cms#25735]
### Steps to reproduce the issue

*See the System - HTTP Headers plugin.*

It is not possible to disable the **X-Content-Type-Options** functionality on an **Apache server**, without changing the **.htaccess** file too.
This is somewhat confusing.

### System information (as much as possible)

PHP Version	7.3.7
Web Server	Apache/2.4.39 (Win64) OpenSSL/1.1.1c PHP/7.3.7
WebServer to PHP Interface	apache2handler
Joomla! Version	Joomla! 4.0.0-alpha11-dev Development [ Amani ] 29-June-2019 11:27 GMT

### Additional comments

Because the **htaccess file** contains namely the following instruction:
```
<IfModule mod_headers.c>
Header always set X-Content-Type-Options "nosniff"
</IfModule>
```

![screen shot 2019-07-29 at 08 11 09](https://issues.joomla.org/uploads/1/bfbaf705cb7161bc61681d6d3da50b6e.png)

**Comment by zero-24** [comment]
>It is not possible to disable the X-Content-Type-Options functionality on an Apache server, without changing the .htaccess file too.
>This is somewhat confusing.

Agree as you mention [this is the line in the htaccess.](https://github.com/joomla/joomla-cms/blob/4.0-dev/htaccess.txt#L24-L27) This was set by the JSST in response to this issue: https://developer.joomla.org/security-centre/766-20190202-core-browserside-mime-type-sniffing-causes-xss-attack-vectors.html

As the option in the plugin was there before it might make sense to remove that option from the plugin.

**Comment by sandewt** [comment]
> Agree as you mention

@zero-24, Yes.

In addition, the **web.config.txt** file has the same X-Content-Type-Options.

See line 29 - 
```
<httpProtocol>
    <customHeaders>
        <add name="X-Content-Type-Options" value="nosniff" />
      </customHeaders>
 </httpProtocol>

```

**Comment by zero-24** [comment]
Yes this is expected, in the coming days I'm preparing an PR to remove that option in the plugin. It is already on my list. :)

**Comment by sandewt** [comment]
@zero-24,

>This was set by the JSST in response...

What in case if you do NOT use the .htaccess or web.config file (the X-Content-Type-Options functionality is then not set, as far as I can oversee) and you *remove* that option?

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25735">issues.joomla.org/tracker/joomla-cms/25735</a>.</sub>

**Comment by zero-24** [comment]
> What in case if you do NOT use the .htaccess or web.config file (the X-Content-Type-Options functionality is then not set, as far as I can oversee) and you remove that option?

What would be your suggestion to fix this? We could force this setting from the plugin but what would happen when they disable the plugin? Every Backed gets a notice and new installs have this header in the default htaccess. Any new installation of 4.0 with htaccess support enabled will have that header set.

**Comment by zero-24** [comment]
Please find here: https://github.com/joomla/joomla-cms/pull/25753 my suggestion to remove that header config from the plugin.

**Comment by ghost** [comment]
Closed as having Pull Request #25753

**Comment by sandewt** [comment]
>What would be your suggestion to fix this? We could force this setting from the plugin ....

Thanks, very clear explanation.

[EDIT] But why not center all the header functionality in the plugin? It must still be possible to remove the code from the htaccess / web.config? Unless the safety is reduced!
